db/document.go

```diff
@@ -491,24 +492,32 @@ func parseXattrStreamData(xattrName string, data []byte) (body []byte, xattr []b
 	for pos < xattrsLen {
 		pairLen := binary.BigEndian.Uint32(data[pos : pos+4])
 		if pairLen == 0 || int(pos+pairLen) > len(data) {
-			return nil, nil, fmt.Errorf("Unexpected xattr pair length (%d) - unable to parse xattrs", pairLen)
+			return nil, nil, nil, fmt.Errorf("Unexpected xattr pair length (%d) - unable to parse xattrs", pairLen)
 		}
 		pos += 4
 		pairBytes := data[pos : pos+pairLen]
 		components := bytes.Split(pairBytes, separator)
 		// xattr pair has the format [key]0x00[value]0x00, and so should split into three components
 		if len(components) != 3 {
-			return nil, nil, fmt.Errorf("Unexpected number of components found in xattr pair: %s", pairBytes)
+			return nil, nil, nil, fmt.Errorf("Unexpected number of components found in xattr pair: %s", pairBytes)
 		}
 		xattrKey := string(components[0])
 		// If this is the xattr we're looking for , we're done
 		if xattrName == xattrKey {
-			return body, components[1], nil
+			xattr = components[1]
 		}
+		if userXattrName == xattrKey {
+			userXattr = components[1]
+		}
+
+		if len(userXattr) > 0 && (len(xattr) > 0 || userXattrName == "") {
```

Comment:
I think this condition is 'exit early if we've collected the xattrs we care about', but it doesn't look correct to me.  If userXattrName == "" it's still waiting for len(userXattr) > 0.  Shouldn't it be `(userXattr>0 || userXattrName=="") && xattr>0`?

Reply:
Yeah I think you're right.